spanner/spanner_arrays/main.go

```diff
@@ -0,0 +1,183 @@
+	defer admin.Close()
+
+	err = createDatabase(ctx, admin, *dsn)
+	if err != nil {
+		log.Fatalf("failed to create database: %v", err)
+	}
+	defer removeDatabase(ctx, admin, *dsn)
+
+	// Connect to database
+	client, err := spanner.NewClient(ctx, *dsn)
+	if err != nil {
+		log.Fatalf("Failed to create client %v", err)
+	}
+	defer client.Close()
+
+	err = loadPresets(ctx, client)
+	if err != nil {
+		log.Fatalf("failed to load preset data: %v", err)
+	}
+
+	it := client.Single().Query(ctx, spanner.NewStatement(`
+		SELECT a.Name AS Name, ARRAY(
+			SELECT b.Name FROM Cities b WHERE a.CountryId = b.CountryId
+		) AS Cities FROM Countries a
+	`))
+	defer it.Stop()
+
+	for {
+		row, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Fatalf("failed to read results: %v", err)
+		}
+
+		var country Country
+		if err = row.ToStruct(&country); err != nil {
+			log.Fatalf("failed to read row into Country struct: %v", err)
+		}
+
+		var cities []string
+		for _, c := range country.Cities {
+			cities = append(cities, c.String())
+		}
+
+		log.Printf("%s: %s", country.Name, strings.Join(cities, ", "))
+	}
+}
+
+func loadPresets(ctx context.Context, db *spanner.Client) error {
+	mx := []*spanner.Mutation{
+		spanner.InsertMap("Countries", map[string]interface{}{
+			"CountryId": 49,
+			"Name":      "Germany",
+		}),
+		spanner.InsertMap("Cities", map[string]interface{}{
+			"CountryId": 49,
+			"CityId":    100,
+			"Name":      "Berlin",
+		}),
+		spanner.InsertMap("Cities", map[string]interface{}{
+			"CountryId": 49,
+			"CityId":    101,
+			"Name":      "Hamburg",
+		}),
+		spanner.InsertMap("Cities", map[string]interface{}{
+			"CountryId": 49,
+			"CityId":    102,
+			"Name":      "Dresden",
+		}),
+		spanner.InsertMap("Countries", map[string]interface{}{
+			"CountryId": 44,
+			"Name":      "United Kingdom",
+		}),
+		spanner.InsertMap("Cities", map[string]interface{}{
+			"CountryId": 44,
+			"CityId":    200,
+			"Name":      "London",
+		}),
+		spanner.InsertMap("Cities", map[string]interface{}{
+			"CountryId": 44,
+			"CityId":    201,
+			"Name":      "Liverpool",
+		}),
+		spanner.InsertMap("Cities", map[string]interface{}{
+			"CountryId": 44,
+			"CityId":    202,
+			"Name":      "Bristol",
+		}),
+		spanner.InsertMap("Cities", map[string]interface{}{
+			"CountryId": 44,
+			"CityId":    203,
+			"Name":      "Newcastle",
+		}),
+	}
+
+	_, err := db.Apply(ctx, mx)
+	return err
+}
+
+func createDatabase(ctx context.Context, adminClient *database.DatabaseAdminClient, db string) error {
+	matches := regexp.MustCompile("^(.*)/databases/(.*)$").FindStringSubmatch(db)
+	if matches == nil || len(matches) != 3 {
+		log.Fatalf("Invalid database id %s", db)
+	}
+
+	var (
+		projectID    = matches[1]
+		databaseName = matches[2]
+	)
+
+	op, err := adminClient.CreateDatabase(ctx, &adminpb.CreateDatabaseRequest{
+		Parent:          databaseName,
+		CreateStatement: fmt.Sprintf("CREATE DATABASE `%s`", projectID),
```

Comment:
Update to be:

CreateStatement: fmt.Sprintf("CREATE DATABASE `%s`", databaseName),